app/forms/signups/create_form.rb

```diff
@@ -0,0 +1,55 @@
+module Signups
+  class CreateForm
+    include ActiveModel::Model
+
+    EN_US_LOCALE = 'en-US'.freeze
+
+    attr_accessor :account_id, :email, :installation_id, :structured_address
+
+    validates :account_id, presence: true
+    validates :installation_id, presence: true
+    validates :email, presence: true, format: { with: URI::MailTo::EMAIL_REGEXP }
+    validate :email_uniqueness
+
+    def submit
+      return unless valid?
+
+      assessment = IncogniaApi.instance.register_signup(**incognia_signup_attrs)
+
+      Signup.create!(
+        account_id: account_id,
+        email: email,
+        address: incognia_signup_attrs[:address]&.to_hash,
+        incognia_signup_id: assessment.id
+      )
+    rescue ActiveRecord::RecordNotUnique
+      errors.add(:email, :taken)
+
+      nil
+    end
+
+    private
+
+    def email_uniqueness
+      errors.add(:email, :taken) if Signup.exists?(email: email)
+    end
+
+    def incognia_signup_attrs
+      return @incognia_signup_attrs if @incognia_signup_attrs
+
+      @incognia_signup_attrs = {
+        installation_id: installation_id,
```

Comment:
> What do you think of using [shorthand hash syntax](https://bugs.ruby-lang.org/issues/14579)?

Creio que o comentário de Herbert se aplique aqui também

Reply:
To avoid conflicts I created a new PR adjusting all places from the most recent branch: https://github.com/inloco/incognia-demo-api-ruby/pull/16

About turning this into a one-line, I just missed. :stuck_out_tongue: